ServerCore/Pages/PuzzleApiController.cs

```diff
@@ -1,65 +1,97 @@
-
-        public PuzzleApiController(PuzzleServerContext context, UserManager<IdentityUser> userManager)
-        {
+namespace ServerCore.Pages
+{
+    /// <summary>
+    /// API controller for puzzles on external sites
+    /// External sites don't have access to auth, so these APIs should rely on data given by this site
+    /// and validate inputs as untrusted
+    /// </summary>
+    public class PuzzleApiController : Controller
+    {
+        private readonly PuzzleServerContext context;
+        private readonly UserManager<IdentityUser> userManager;
+
+        public PuzzleApiController(PuzzleServerContext context, UserManager<IdentityUser> userManager)
+        {
             this.context = context;
-            this.userManager = userManager;
-        }
-
-        /// <summary>
-        /// Allows external puzzles to validate a team password
-        /// </summary>
-        /// <param name="teamPassword">Potential team password</param>
-        /// <returns>True if the password belongs to a team, false otherwise</returns>
-        [HttpGet]
-        [Route("api/puzzleapi/validateteampassword")]
-        public async Task<ActionResult<bool>> GetValidateTeamPasswordAsync(string teamPassword)
-        {
-            if (string.IsNullOrWhiteSpace(teamPassword))
-            {
-                return false;
-            }
-
-            return await (from team in context.Teams
-                          where team.Password == teamPassword
-                          select team).AnyAsync();
-        }
-
+            this.userManager = userManager;
+        }
+
+        /// <summary>
+        /// Allows external puzzles to validate a team password
+        /// </summary>
+        /// <param name="teamPassword">Potential team password</param>
+        /// <returns>True if the password belongs to a team, false otherwise</returns>
+        [HttpGet]
+        [Route("api/puzzleapi/validateteampassword")]
+        public async Task<ActionResult<bool>> GetValidateTeamPasswordAsync(string teamPassword)
+        {
+            if (string.IsNullOrWhiteSpace(teamPassword))
+            {
+                return false;
+            }
+
+            return await (from team in context.Teams
+                          where team.Password == teamPassword
+                          select team).AnyAsync();
+        }
+
+        /// <summary>
+        /// Allows external puzzles to validate a team password
+        /// </summary>
+        /// <param name="teamPassword">Potential team password</param>
+        /// <returns>True if the password belongs to a team, false otherwise</returns>
+        [HttpGet]
+        [Route("api/puzzleapi/getmailinfo")]
+        public async Task<ActionResult<MailInfo>> GetMailInfoAsync(string eventId, int puzzleId)
+        {
+            Event currentEvent = await EventHelper.GetEventFromEventId(context, eventId);
+            Team team = await UserEventHelper.GetTeamForCurrentPlayer(context, currentEvent, User, userManager);

```

Comment:
What happens if the player isn't on a team in these scenarios?
1. Not signed up for this event
2. Author
3. Admin

Reply:
Now verifies that you're on a team. No special handling of authors/admins, no need for such things.